File: FE/lib/payload/request/trip_request.dart
class TripRequest {
  // Cập nhật mới: Thêm customerId để Backend biết ai đặt
  final String customerId;

  final double pickupLatitude;
  final double pickupLongitude;
  final String pickupAddress;

  final String? destinationAddress;
  final double? destinationLatitude;
  final double? destinationLongitude;

  final String vehicleType; // "BIKE", "CAR_4", "CAR_7"

  TripRequest({
    required this.customerId,
    required this.pickupLatitude,
    required this.pickupLongitude,
    required this.pickupAddress,
    this.destinationAddress,
    this.destinationLatitude,
    this.destinationLongitude,
    required this.vehicleType,
  });

  Map<String, dynamic> toJson() {
    return {
      'customerId': customerId,
      'pickupLatitude': pickupLatitude,
      'pickupLongitude': pickupLongitude,
      'pickupAddress': pickupAddress,
      'destinationAddress': destinationAddress,
      'destinationLatitude': destinationLatitude,
      'destinationLongitude': destinationLongitude,
      'vehicleType': vehicleType,
    };
  }
}

File: taxi-api/src/main/java/com/smarttaxi/taxi_api/payload/request/TripRequest.java
package com.smarttaxi.taxi_api.payload.request;

public class TripRequest {
    
    private Double pickupLatitude;
    private Double pickupLongitude;
    private String pickupAddress; // Đã thêm trường này
    
    private String destinationAddress;
    private Double destinationLatitude;
    private Double destinationLongitude;
    
    private String vehicleType;

    // --- GETTERS & SETTERS (Viết tay) ---

    public Double getPickupLatitude() { return pickupLatitude; }
    public void setPickupLatitude(Double pickupLatitude) { this.pickupLatitude = pickupLatitude; }

    public Double getPickupLongitude() { return pickupLongitude; }
    public void setPickupLongitude(Double pickupLongitude) { this.pickupLongitude = pickupLongitude; }

    public String getPickupAddress() { return pickupAddress; }
    public void setPickupAddress(String pickupAddress) { this.pickupAddress = pickupAddress; }

    public String getDestinationAddress() { return destinationAddress; }
    public void setDestinationAddress(String destinationAddress) { this.destinationAddress = destinationAddress; }

    public Double getDestinationLatitude() { return destinationLatitude; }
    public void setDestinationLatitude(Double destinationLatitude) { this.destinationLatitude = destinationLatitude; }

    public Double getDestinationLongitude() { return destinationLongitude; }
    public void setDestinationLongitude(Double destinationLongitude) { this.destinationLongitude = destinationLongitude; }

    public String getVehicleType() { return vehicleType; }
    public void setVehicleType(String vehicleType) { this.vehicleType = vehicleType; }
}